File: 01-Investigacion/CSharp/fibonacci.cs
using System.Diagnostics;

class Program
{
    static List<ulong> GenerarFibonacci(int n)
    {
        List<ulong> secuenciaFibonacci = new();  // Inicio con una lista vacía
        ulong a = 0, b = 1;  // Establezco los dos primeros números de la secuencia Fibonacci
        for (int i = 0; i < n; i++) // Realizo un bucle que se ejecutará 'n' veces
        {
            secuenciaFibonacci.Add(a);  // Añado el número actual 'a' a la lista
            ulong temp = a;
            a = b;
            b = temp + b;  // Actualizo 'a' y 'b' siendo la siguiente secuencia la suma de las ultimas dos
        }
        return secuenciaFibonacci;  // Devuelvo la lista con los números de Fibonacci
    }

    static void Main()
    {
        Stopwatch stopwatch = new();
        stopwatch.Start();
        List<ulong> secuenciaFibonacci = GenerarFibonacci(100000); // Llamo a la función para generar 100,000 números de Fibonacci
        stopwatch.Stop();
        // Console.WriteLine(string.Join(", ", secuenciaFibonacci));
        Console.WriteLine("Tiempo de ejecución en milisegundos: " + stopwatch.ElapsedMilliseconds); // Imprimo el tiempo de ejecución total en milisegundos
    }
}

File: 01-Investigacion/Python/fibonacci.py
import time # Importo la biblioteca time para medir el tiempo de ejecución

def generar_fibonacci(n):
    secuencia_fibonacci = [] # Comienzo con una lista vacia
    a, b = 0, 1 # Utilizo los primeros 2 numeros de la secuencia
    for _ in range(n): # Genera n numeros de fibonacci
        secuencia_fibonacci.append(a) # Añado el número actual 'a' a la lista
        a, b = b, a + b # Actualizo los valores de 'a' y 'b' siendo b = a + b
    return secuencia_fibonacci

if __name__ == "__main__":
    start_time = time.time()
    fibonacci_numbers = generar_fibonacci(100000) # Genera una lista de 100.000 numeros de fibonacci
    end_time = time.time()
    # print(fibonacci_numbers)
    print("Tiempo de ejecución en segundos:", end_time - start_time)
